cmd/status/status.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/jaegertracing/jaeger/cmd/flags"
+	"github.com/jaegertracing/jaeger/ports"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// Command for running status check against the AdminPort.
+func Command(v *viper.Viper, adminPort int) *cobra.Command {
+	adminServer := flags.NewAdminServer(ports.PortToHostPort(adminPort))
+	adminURL := ""
+	c := &cobra.Command{
+		Use:   "status",
+		Short: "Print the status.",
+		Long:  `Prints admin status information, exit non-zero on any error.`,
+		PreRunE: func(cmd *cobra.Command, args []string) (err error) {
+			rootCmd := cmd
+			for rootCmd.Parent() != nil {
+				rootCmd = rootCmd.Parent()
+			}
```

Comment:
what is the objective of this loop?

Reply:
Good question. This was left over from the initial commit where I selected the correct part based on the name of the command. So this would give me 'jaeger-agent'. To be fair, it's a bit cargo culty, as I looked at how the docs subcommand (I believe, on mobile can't check) worked and emulated it's behavior.

Reply:
it doesn't look like it's even used

Reply:
I was using it to find the flags or try to figure out where they were being set. I don't believe it has much value now or won't by the time it all functions.